extensions/amp-font/0.1/amp-font.js

```diff
@@ -0,0 +1,316 @@
+      this.loadFontWithNativeApis_();
+    } else {
+      this.onBeforeForceLoadFont_();
+      this.forceLoadFont_();
+    }
+  }
+
+
+  /**
+   * If the browser supports font loading events use them to load the font.
+   * @private
+   */
+  loadFontWithNativeApis_() {
+    /* style | variant | weight | size/line-height | family */
+    /* font: italic small-caps bolder 16px/3 cursive; */
+    let fontString = (
+      this.customFontStyle_ + ' ' +
+      this.customFontVariant_ + ' ' +
+      this.customFontWeight_ + ' ' +
+      this.customFontSize_ + ' ' +
+      this.customFontFamily_);
+    if (this.document_.fonts.check(fontString)) {
+      this.onFontLoadSuccess_();
+    } else {
+      this.document_.fonts.load(fontString).then(() => {
+        this.onFontLoadSuccess_();
+      });
+    }
+  }
+
+
+  /**
+   * Step 1 for loading font on browsers that don't support font loading events.
+   * Creates divs hidden from the viewport and measures dimensions for default
+   * fonts.
+   * @private
+   */
+  onBeforeForceLoadFont_() {
+    var ncustomFontFamily_ = normalizeFontFamily_(this.customFontFamily_);
+    var containerElement = this.document_.createElement('div');
+    containerElement.setAttribute(
+      'id', '-amp-font-container-' + ncustomFontFamily_);
+    style.setStyles(containerElement, {
+      fontSize: '48px',
+      fontVariant: 'normal',
+      left: '-999px',
+      lineHeight: 'normal',
+      margin: 0,
+      padding: 0,
+      position: 'absolute',
+      top: '-999px'
+    });
+    DEFAULT_FONTS_.forEach(font => {
+      let defaultFontElement = this.document_.createElement('div');
+      let normalizedFont = normalizeFontFamily_(font);
+      let defaultFontElementId =
+          '-amp-font-' + ncustomFontFamily_ + '-' + normalizedFont;
+      defaultFontElement.setAttribute('id', defaultFontElementId);
+      defaultFontElement.textContent = TEST_STRING_;
+      style.setStyles(defaultFontElement, {
+        fontFamily: font,
+        margin: 0,
+        padding: 0,
+        whiteSpace: 'nowrap'
+      });
+      containerElement.appendChild(defaultFontElement);
+    });
+    this.document_.body.appendChild(containerElement);
+    DEFAULT_FONTS_.forEach(font => {
+      let normalizedFont = normalizeFontFamily_(font);
+      let defaultFontElementId =
+          '-amp-font-' + ncustomFontFamily_ + '-' + normalizedFont;
+      let element = this.document_.getElementById(defaultFontElementId);
+      this.measurements_[defaultFontElementId] = {
+        width: element./*OK*/offsetWidth,
+        height: element./*OK*/offsetHeight
+      };
+    });
+  }
+
+
+  /**
+   * Make the browser that don't support font loading events to download the
+   * custom font by creating an element (with text) not visible on the viewport.
+   * font download is detected by comparing the elements height and width with
+   * measurements taken in step1.
+   * @private
+   */
+  forceLoadFont_() {
```

Comment:
The name is a bit misleading. Maybe `checkIfFontLoaded_`? Or something monitory...